web-components/src/components/menu-overlay/scss/menu-overlay.scss

```diff
@@ -48,28 +46,64 @@
   .overlay-container[data-popper-placement^="top"] > .overlay-arrow {
     border-left: solid $arrow-size transparent;
     border-right: solid $arrow-size transparent;
-    border-top: solid $arrow-size var(--md-menu-overlay-arrow-bg-color, var(--md-tertiary-white-bg-color));
+    border-top: solid $arrow-size var(--md-menu-overlay-border-color, --gray-20);
     bottom: $arrow-offset;
+    filter: drop-shadow(0px 14px 8px rgba(0, 0, 0, 0.16));
   }
 
   .overlay-container[data-popper-placement^="bottom"] > .overlay-arrow {
-    border-bottom: solid $arrow-size var(--md-menu-overlay-arrow-bg-color, var(--md-tertiary-white-bg-color));
+    border-bottom: solid $arrow-size var(--md-menu-overlay-border-color, --gray-20);
     border-left: solid $arrow-size transparent;
     border-right: solid $arrow-size transparent;
     top: $arrow-offset;
   }
 
   .overlay-container[data-popper-placement^="right"] > .overlay-arrow {
     border-bottom: solid $arrow-size transparent;
-    border-right: solid $arrow-size var(--md-menu-overlay-arrow-bg-color, var(--md-tertiary-white-bg-color));
+    border-right: solid $arrow-size var(--md-menu-overlay-border-color, --gray-20);
     border-top: solid $arrow-size transparent;
     left: $arrow-offset;
+    filter: drop-shadow(-16px 0px 10px rgba(0, 0, 0, 0.16));
   }
 
   .overlay-container[data-popper-placement^="left"] > .overlay-arrow {
     border-bottom: solid $arrow-size transparent;
-    border-left: solid $arrow-size var(--md-menu-overlay-arrow-bg-color, var(--md-tertiary-white-bg-color));
+    border-left: solid $arrow-size var(--md-menu-overlay-border-color, --gray-20);
     border-top: solid $arrow-size transparent;
     right: $arrow-offset;
+    filter: drop-shadow(16px 0px 10px rgba(0, 0, 0, 0.16));
   }
-}
+
+  .overlay-container > .overlay-arrow:after {
+    content: "";
+    position: absolute;
+    border-width: 19px;
+    border-style: solid;
+    border-color: transparent;
+  }
+
+  // styling to create triangle border to match menu for pointer
```

Comment:
Below just creates a menu pointer with the same color as the border color. This allows us to overlay the white menu pointer and offset it by 2 pixels given us an arrow with a seamless border:
<img width="563" alt="image" src="https://github.com/user-attachments/assets/c4d8991b-252d-4816-b2d8-4bfd244210d5">